modules/gapi/src/backends/oak/goakbackend.cpp

```diff
@@ -0,0 +1,691 @@
+                              typename cv::detail::MkSeq<sizeof...(Outs)>::type());
+    }
+};
+
+} // namespace detail
+
+struct GOAKKernel {
+    using F = std::function<std::shared_ptr<dai::Node>(GOAKContext&,
+                                                       std::vector<std::pair<std::string, dai::Buffer>>&)>;
+    explicit GOAKKernel(const F& f) : m_f(f) {}
+    const F m_f;
+};
+
+struct OAKComponent
+{
+    static const char *name() { return "OAK Component"; }
+    GOAKKernel k;
+};
+
+}} // namespace gimpl // namespace cv
+
+using OAKGraph = ade::TypedGraph
+    < cv::gimpl::OAKComponent
+    // FIXME: extend
+    >;
+
+using ConstOAKGraph = ade::ConstTypedGraph
+    < cv::gimpl::OAKComponent
+    // FIXME: extend
+    >;
+
+// This function links dai operation nodes - parent's output to child's input.
+// It utilizes G-API graph to search for operation's node it's previous operation in graph
+// when links them in dai graph.
+void cv::gimpl::GOAKExecutable::LinkToParentHelper(ade::NodeHandle handle,
+                                                   const std::vector<ade::NodeHandle>& nodes)
+{
+    ade::NodeHandle parent;
+    for (const auto& data_nh : handle.get()->inNodes()) {
+        // data_nh - node's input data
+        // need to find which other node produces that data
+        for (const auto& nh : nodes) {
+            if (m_gm.metadata(nh).get<NodeType>().t == NodeType::OP) {
+                for (const auto& outdatah : nh.get()->outNodes()) {
+                    if (data_nh == outdatah) {
+                        parent = nh;
+                    }
+                }
+            }
+        }
+    }
+    GAPI_Assert(m_oak_nodes.find(parent) != m_oak_nodes.end());
+    GAPI_Assert(m_oak_nodes[handle].inputs.size() ==
+                m_oak_nodes[parent].outputs.size());
+    for (size_t i = 0; i < m_oak_nodes[handle].inputs.size(); ++i) {
+        m_oak_nodes[parent].outputs[i]->link(*(m_oak_nodes[handle].inputs[i]));
+    }
+}
+
+std::reference_wrapper<cv::GArg>
+cv::gimpl::GOAKExecutable::packArg(const GArg &arg,
+                                   std::vector<ExtractTypeHelper::InputPtr>& oak_ins,
+                                   std::vector<cv::GArg>& oak_wrapped_args) {
+    if (arg.kind != cv::detail::ArgKind::GOBJREF) {
+        GAPI_Assert(   arg.kind != cv::detail::ArgKind::GMAT
+                    && arg.kind != cv::detail::ArgKind::GSCALAR
+                    && arg.kind != cv::detail::ArgKind::GARRAY
+                    && arg.kind != cv::detail::ArgKind::GOPAQUE
+                    && arg.kind != cv::detail::ArgKind::GFRAME);
+        // All other cases - pass as-is, with no transformations to
+        // GArg contents.
+        return const_cast<cv::GArg&>(arg);
+    }
+    const cv::gimpl::RcDesc &ref = arg.get<cv::gimpl::RcDesc>();
+    switch (ref.shape) {
+    case GShape::GFRAME:
+        oak_ins.push_back(nullptr);
+        oak_wrapped_args.push_back(GArg(oak_ins.back()));
```

Comment:
I believe you wanted to put a reference on oak_ins.back() here, but made a copy instead

Reply:
Nice catch, fixed, thanks